src/erlcloud_kinesis.erl

```diff
@@ -392,36 +392,51 @@ normalize_response([]) -> [].
 %% @end
 %%------------------------------------------------------------------------------
 
--spec put_record/3 :: (string(), string(), string()) -> proplist().
+-type partition_key()     :: binary().
+-type payload()           :: binary() | string().
+-type explicit_hash_key() :: binary() | undefined.
+-type ordering()          :: binary() | undefined.
 
+-spec put_record/3 :: (binary(), partition_key(), payload()) ->
+    {ok, proplist()} | {error, any()}.
 put_record(StreamName, PartitionKey, Data) ->
-  Json = [{<<"StreamName">>, StreamName}, {<<"PartitionKey">>, PartitionKey}, {<<"Data">>, base64:encode(Data)}],
-  erlcloud_kinesis_impl:request(default_config(), "Kinesis_20131202.PutRecord", Json).
-
--spec put_record/4 :: (string(), string(), string(), string() | aws_config()) -> proplist().
+    put_record(StreamName, PartitionKey, Data, undefined).
```

Comment:
Are you sure to change function interface from string() to binary()?
Not sure it is safe to do so.